wsd/DocumentBroker.cpp

```diff
@@ -3340,38 +3340,23 @@ void DocumentBroker::sendRequestedTiles(const std::shared_ptr<ClientSession>& se
 
     size_t tilesOnFlyUpperLimit = session->getTilesOnFlyUpperLimit();
 
-    // Drop tiles which we are waiting for too long
-    session->removeOutdatedTilesOnFly();
-
     auto now = std::chrono::steady_clock::now();
 
+    // Drop tiles which we are waiting for too long
+    session->removeOutdatedTilesOnFly(now);
+
     // All tiles were processed on client side that we sent last time, so we can send
     // a new batch of tiles which was invalidated / requested in the meantime
     std::deque<TileDesc>& requestedTiles = session->getRequestedTiles();
     if (!requestedTiles.empty() && hasTileCache())
     {
-        std::size_t delayedTiles = 0;
         std::vector<TileDesc> tilesNeedsRendering;
         bool allSamePartAndSize = true;
-        std::size_t beingRendered = _tileCache->countTilesBeingRenderedForSession(session, now);
-        while (session->getTilesOnFlyCount() + beingRendered < tilesOnFlyUpperLimit &&
-              !requestedTiles.empty() &&
-              // If we delayed all tiles we don't send any tile (we will when next tileprocessed message arrives)
-              delayedTiles < requestedTiles.size())
+        while (!requestedTiles.empty() &&
+               session->getTilesOnFlyCount() < tilesOnFlyUpperLimit)
         {
             TileDesc& tile = *(requestedTiles.begin());
 
-            // We already sent out two versions of the same tile, let's not send the third one
-            // until we get a tileprocessed message for this specific tile.
-            if (session->countIdenticalTilesOnFly(tile) >= 2)
```

Comment:
oddly enough TileCacheTests::testLimitTileVersionsOnFly apparently tries to test for this but seeing as ci is green the test didn't catch this change